### PR TITLE
doc: input: add a list of general purpose drivers

### PR DESCRIPTION
**Pull request by fabiobaltieri**
Add a list of general purpose input drivers, hopefully this will make them easier to be discovered by users.

### CODE REVIEW

**Review by gmarull — DISMISSED** [review]

**Review by faxe1008 — DISMISSED** [review]
Good idea, I have seen new Zephyr users reinvent the wheel with some self built button debouncing :^)

**Comment by fabiobaltieri** [comment]
> Yeah, heard about this in an embedded.fm episode as well, in fairness right now the only way to find these now is to go through the code.

AKA: the RIGHT way. :-)

**Review by bjarki-andreasen — DISMISSED** [review]

File: doc/services/input/index.rst [hunk 1]
```diff
@@ -86,8 +86,28 @@ compatibility device node, for example:
...
+- :dtcompatible:`gpio-qdec`: for GPIO connected quadrature encoders.
+- :dtcompatible:`input-keymap`: maps row/col/touch events from a keyboard
+  matrix to key events.
+- :dtcompatible:`zephyr,input-longpress`: listens for key events, emits events
+  for short and long press.
+- :dtcompatible:`zephyr,lvgl-button-input`
+  :dtcompatible:`zephyr,lvgl-encoder-input`
+  :dtcompatible:`zephyr,lvgl-keypad-input`
+  :dtcompatible:`zephyr,lvgl-pointer-input`: listens for input events and
+  translates those to various type of LVGL input devices.
```
Comment (kartben) [inline doc/services/input/index.rst:107]:
```suggestion
  translates those to various types of LVGL input devices.
```

File: doc/services/input/index.rst [hunk 2]
```diff
@@ -86,8 +86,28 @@ compatibility device node, for example:
...
+General Purpose Drivers
+***********************
+
+- :dtcompatible:`adc-keys`: for buttons connected to a resistor ladder.
+- :dtcompatible:`analog-axis`: for absolute position devices connected to an
+  ADC input (thumbsticks, sliders...).
+- :dtcompatible:`gpio-kbd-matrix`: for GPIO connected keyboard matrices.
+- :dtcompatible:`gpio-keys`: for switches directly connected to a GPIO,
+  implements button debouncing.
+- :dtcompatible:`gpio-qdec`: for GPIO connected quadrature encoders.
```
Comment (kartben) [inline doc/services/input/index.rst:98]:
```suggestion
- :dtcompatible:`gpio-qdec`: for GPIO-connected quadrature encoders.
```

File: doc/services/input/index.rst [hunk 3]
```diff
@@ -86,8 +86,28 @@ compatibility device node, for example:
...
 
-Driver Documentation
-********************
+General Purpose Drivers
+***********************
+
+- :dtcompatible:`adc-keys`: for buttons connected to a resistor ladder.
+- :dtcompatible:`analog-axis`: for absolute position devices connected to an
+  ADC input (thumbsticks, sliders...).
+- :dtcompatible:`gpio-kbd-matrix`: for GPIO connected keyboard matrices.
```
Comment (kartben) [inline doc/services/input/index.rst:95]:
```suggestion
- :dtcompatible:`gpio-kbd-matrix`: for GPIO-connected keyboard matrices.
```

**Review by kartben — CHANGES_REQUESTED** [review]

File: doc/services/input/index.rst (hunk 1 above)
Comment (fabiobaltieri) [reply doc/services/input/index.rst]: done

File: doc/services/input/index.rst (hunk 3 above)
Comment (fabiobaltieri) [reply doc/services/input/index.rst]: done

File: doc/services/input/index.rst (hunk 2 above)
Comment (fabiobaltieri) [reply doc/services/input/index.rst]: done

**Review by kartben — APPROVED** [review]

**Review by bjarki-andreasen — APPROVED** [review]

**Pull request merged**